app-android/src/main/java/com/ustadmobile/port/android/view/binding/DatePickerBindingAdapter2.kt

```diff
@@ -0,0 +1,277 @@
+ * a default. 0 and Long.MAX_VALUE are reserved defaults. MAX_VALUE is used for end times to simplify
+ * queries.
+ */
+val Long.isSet2: Boolean
+    get(){
+        return (
+                (this < -43200000 || this > 43200000)  &&
+                (this < (Long.MAX_VALUE - 43200000)  )
+                )
+    }
+
+@BindingAdapter(value = ["dateTimeInMillis", "timeZoneId", "dateTimeInMillisMode"])
+fun TextView.setDateTime2(timeInMillis: Long, timeZoneId: String?, dateTimeInMillisMode: Int) {
+    val timeZone = TimeZone.getTimeZone(timeZoneId?:"UTC")
+    val formattedTimeInMillis = if (timeInMillis == Long.MAX_VALUE){
+        Long.MAX_VALUE - 43200000
+    }else{
+        timeInMillis
+    }
+    adapterCalendar.also {
+        it.timeZone = timeZone
+        it.timeInMillis = formattedTimeInMillis
+
+        if(dateTimeInMillisMode == MODE_START_OF_DAY) {
+            it.set(Calendar.HOUR_OF_DAY, 0)
+            it.set(Calendar.MINUTE, 0)
+            it.set(Calendar.SECOND, 0)
+            it.set(Calendar.MILLISECOND, 0)
+        }else {
+            it.set(Calendar.HOUR_OF_DAY, 23)
+            it.set(Calendar.MINUTE, 59)
+            it.set(Calendar.SECOND, 59)
+            it.set(Calendar.MILLISECOND, 59)
+        }
+    }
+    adapterDateFormat.timeZone = timeZone
+
+    val impl : UstadMobileSystemImpl = (context.applicationContext as DIAware
+            ).di.direct.instance()
+    val localeString = impl.getLocale(context).toString()
+    val persianDate = PersianDateImpl()
+    persianDate.setDate(adapterCalendar.timeInMillis)
+
+    text = if(!adapterCalendar.timeInMillis.isSet2){
+        ""
+    }else{
+        if(localeString.startsWith("ps") ||
+            localeString.startsWith("fa")
+        ){
+            persianDate.persianYear.toString() + "/" +
+                    persianDate.persianMonth + "/" +
+                    persianDate.persianDay
+        }else {
+            adapterDateFormat.format(Date(adapterCalendar.timeInMillis))
+        }
+    }
+}
+
+@InverseBindingAdapter(attribute = "dateTimeInMillis")
+fun TextView.getDateTimeInMillis(): Long {
+    return adapterCalendar.timeInMillis
+}
+@BindingAdapter("dateTimeInMillisAttrChanged")
+fun TextView.setDateTimeInMillisChanged(inverseBindingListener: InverseBindingListener) {
+    setOnClickListener {
+        val dialogView = LayoutInflater.from(context).inflate(R.layout.dialog_date_picker,
+            null, false)
+
+        val cal = adapterCalendar
+        val defaultCal = if(!adapterCalendar.timeInMillis.isSet2){
+            Calendar.getInstance()
+        }else{
+            cal
+        }
+
+        val impl : UstadMobileSystemImpl = (context.applicationContext as DIAware
+                                            ).di.direct.instance()
+        val localeString = impl.getLocale(context)
+
+        if(localeString.startsWith("ps") || localeString.startsWith("fa")){
+
+            val picker: PersianDatePickerDialog = PersianDatePickerDialog(context)
```

Comment:
This date picker will not be connected to the time zone. This could use the PersianCalendar that comes with the library which will accept a timezone or ( https://github.com/amirmehdizadeh/JalaliCalendar )